### DISCUSSION TITLE
Color codes should not count for the character limit when setting a nickname

### DISCUSSION THREAD

**Discussion by DarkKingKaan** [EssentialsX/Essentials#4245] [Ideas and Feature Suggestions] [answered]
There's an option in the config that allows us to set a maximum number of symbols/characters when creating a nickname with the command "/nick". I think that in case someone uses a colour code, the code part should not be counted as a character in the nickname.

Now it looks like this:
`/nick &a&lMyNickname` --> 14 characters

The new system would make it count only 10 characters since the "&a&l" part wouldn't be counted, as it's only a colour code that is not visible in the chat.

People on my server are sometimes using the new colour codes that use the HEX system, therefore I had to set the limit much higher, exposing myself to some people "abusing" this by setting very, very long names. I think this would solve this issue for many people.

If you have a comment to make, feel free to do it. If you have any more suggestions/additions to my idea, please let me know.

Thanks!

**Comment by tevzselcan** [comment]
Yes, I agree with this, as I have had the same problem with this before. Good sugestion :)

**Comment by triagonal** [comment, marked as answer]
This is already a config option: `ignore-colors-in-max-nick-length`

https://github.com/EssentialsX/Essentials/blob/780aadedf2fe585257852be26cb179760396285e/Essentials/src/main/resources/config.yml#L41-L43

**Reply by DarkKingKaan** [reply]
Didn't notice this, thanks!
